File: 1.两数之和.cpp
/*
 * @lc app=leetcode.cn id=1 lang=cpp
 *
 * [1] 两数之和
 */

// @lc code=start
#include <vector>
#include <unordered_map>
using namespace std;
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int len =  nums.size();
        unordered_map<int, int> res = {};
        vector<int> fuck;
        int tmp = 0;
        for(int i = 0; i < len; i++){
            tmp =  target - nums[i];
            if(res.find(tmp) != res.end()){
                return {res[tmp], i};
            }
            res[nums[i]] = i;


        }
        return {};
    }
};
// @lc code=end

File: src/code_without_input/1.两数之和.java
/*
 * @lc app=leetcode.cn id=1 lang=java
 *
 * [1] 两数之和
 */

// @lc code=start

import java.util.HashMap;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> tmp = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            if(tmp.containsKey(target - nums[i])){
                return new int[]{i, tmp.get(target - nums[i])};
            }else{
                tmp.put(nums[i], i);
            }
        }
        return null;
    }
}
// @lc code=end

